app/src/main/java/net/gaast/giggity/Schedule.java

```diff
@@ -512,39 +521,45 @@ private void loadJson(BufferedReader in) {
 					item.addLink(new Link(event.getString("signup_url")));
 				}
 
-				JSONObject microlocation = event.getJSONObject("microlocation");
-				String location = microlocation.getString("name");
 
-				if ((line = tentMap.get(location)) == null) {
-					line = new Schedule.Line(location, location);
-					tents.add(line);
-					tentMap.put(location, line);
-				}
+                JSONObject microlocation = event.getJSONObject("microlocation");
+                String location = microlocation.getString("name");
+
+                if ((line = tentMap.get(location)) == null) {
+                    line = new Schedule.Line(location, location);
+                    tents.add(line);
+                    tentMap.put(location, line);
+                }
+
 
 				//Getting value (latitude and longitude) from the map by key (name)
-				String locString = locs.get(line.getTitle());
-				String latitude = locString.substring(0, locString.indexOf(','));
-				String longitude = locString.substring(locString.indexOf(',') + 1);
 
-				//Adding location details here
-				String latlon = null;
-				try {
-					latlon = ("geo:0,0?q=" + longitude + "," +
-							latitude + "(" +
-							URLEncoder.encode(line.getTitle(), "utf-8") + ")");
-				} catch (UnsupportedEncodingException e) {
-					e.printStackTrace();
-				}
+                if(hasMicrolocs) {
+                    String locString = locs.get(line.getTitle());
+                    String latitude = locString.substring(0, locString.indexOf(','));
+                    String longitude = locString.substring(locString.indexOf(',') + 1);
 
+                    //Adding location details here
+                    String latlon = null;
+                    try {
+                        latlon = ("geo:0,0?q=" + longitude + "," +
+                                latitude + "(" +
+                                URLEncoder.encode(line.getTitle(), "utf-8") + ")");
+                    } catch (UnsupportedEncodingException e) {
+                        e.printStackTrace();
+                    }
 
-				if (latlon != null) {
-					line.setLocation(latlon);
-				}
+
+                    if (latlon != null) {
+                        line.setLocation(latlon);
+                    }
+                }
 
 				line.addItem(item);
 			}
 
 		} catch (JSONException e) {
+            Log.w("JSON parsing", " Unable to parse ", e);
```

Comment:
Log.w isn't really any more helpful than e.printStrackTrace() :-/ See the LoadException thing above. That should make it to the user properly.